src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,93 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        String lineSeparator = System.lineSeparator();
+
+        result.append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(lineSeparator);
+
+        int[] salaries = this.countSalaries(names, data, dateFrom, dateTo);
+
+        for (int i = 0; i < names.length; i++) {
+            result.append(names[i])
+                    .append(" - ")
+                    .append(salaries[i]);
+            if (i != names.length - 1) {
+                result.append(lineSeparator);
+            }
+        }
+
+        return result.toString();
+    }
```

Comment:
iterate over names, then inside that loop iterate over data for each name, and count salary
```
for names {
  int salary
  for data {
      if data[NAME_INDEX] equals name
         salary += hours * per hours
  }
  append report
}
```

Reply:
I've already done this in the separate method "countSalaries".

Reply:
In "getSalaryInfo" method I combine names and salaries into StringBuilder.

Reply:
I agree with Anastasia. Your class looks very complicated better use the approach she suggested. It should be two nested loops inside one method approximately 30 lines code.

Reply:
I have a question.
Should I keep "isDateValid" method or this this the wrong way to validate the date?
Because we need to count salaries only for the specific period and we need to validate that this date is in the period.
@Rommelua @aaananas